### DISCUSSION TITLE
试用后的一些想法和建议

### DISCUSSION THREAD

**Discussion by Oloong** [hhyo/Archery#1638] [General]
1.数据字典，过于简单，建议开发人员可以纠错，动态维护。
a.开发人员发现问题，提出修改工单，DBA审核后更新。
b.mysql的数据字典不要从information_schema表中获取，因为无法编辑，但可做初始化参照。需在archery库中另外建表维护。

2.执行异常的工单，非申请人的问题导致的异常，不应该直接终止，DBA介入处理后，正常执行，可让原工单流程正常结束。

3.关于SQL备份，建议加入间接备份机制。不能自动备份的如分库分表的SQL，可在执行前查询好结果集生成SQL,间接备份。

4.结果集导出功能问题
a.NULL值全部成了（NULL） 字符串
b.大数字全部成了科学计算法展示，建议数字转成字符串处理
c.大数据集导出有性能问题，同步改队列异步，严格控制并发。建议采用异步生成文件，再下载的方式
d.另外可添加一个导出工单流程，提交SQL，由DBA导出交付

5.流程问题
SQL审核->SQL上线->点击一个工单->查看提交信息  
查看提交信息页面，应禁止在此处可提交订单！

6.重复的工单，系统无法智能提示，容易造成误操作。如回滚工单，可以一直回滚下去。

7.系统设置，只有全局化，少了实例或资源组成面的配置，不能针对具体实例或组做特殊化配置
   比如有些实例，如开发实例或测试环境，可以让开发或测试人员自动审核，自行执行SQL等。
 
写这么多先

**Comment by LeoQuote** [comment]
修改工单确实是一个很需要的需求, 不过也需要确认不会造成审批时和通过时的工单内容不同的情况. 这个后续可以通过开发来解决

执行异常的工单打回重新执行, 或者打回修改后重新走审批, 这都需要审批流的大改, 目前审批流没有办法做到能回到上一节点, 我记得是这样的., 开发可能还是需要一定精力.

SQL 备份的完全是 goinception 的功能, 你说的这个方法是一个方法, 但是备份完了之后回滚语句还是要自己去转写, 都不是很容易, 目前archery 没 SQL 引擎, 改写这个可以说是非常难了.

结果集导出问题, 这个完全由前端组件提供, 你可以导出两个问题都是 excel 的相关问题, 你可以导出成 csv 应该能解决问题

大数据导出本身就有数据安全风险, 这部分我们实际上应该是不会做, archery 定位不是这样的大数据导出的.

查看提交页面做的时候应该是复用了之前的东西, 那这块应该是偷懒了, 可以做一个单独的 model 或者view 来解决.

根据实例或资源组的单独配置, 这个可能可以做, 配置表可以有一个 scope 字段, 根据 scope 来取对应的配置.


欢迎 pr 哈.

**Comment by hydgladiator** [comment]
“工单审核流配置”这种配置型的，能否做成模版，这样如果有多个资源组，可以在模版的基础上做细微的调整？
